src/net/server/message.rs

```diff
@@ -0,0 +1,372 @@
+//! Support for working with DNS messages in servers.
+use core::{ops::ControlFlow, sync::atomic::Ordering};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use tokio::task::JoinHandle;
+use tracing::info_span;
+
+use crate::{
+    base::{message::ShortMessage, Message},
+    net::server::{
+        buf::BufSource, metrics::ServerMetrics,
+        middleware::chain::MiddlewareChain,
+    },
+};
+
+use super::service::{
+    CallResult, Service, ServiceError, ServiceResultItem, Transaction,
+};
+
+//------------ MsgProvider ---------------------------------------------------
+
+/// A MsgProvider can determine the number of bytes of message data to expect
+/// and then turn those bytes into a concrete message type.
+pub trait MsgProvider<RequestOctets: AsRef<[u8]>> {
```

Comment:
I thought this was already removed. It was originally created back when I thought the same TCP stream server could also be used for non-DNS purposes but I rejected that approach because things like disconnect with flush or not handling are DNS RFC specific. I thought I had already removed this, perhaps I did in some changes that I threw away and this got left behind. I'll remove it.